forge/src/main/java/net/blay09/mods/balm/forge/world/ForgeBalmWorldGen.java

```diff
@@ -76,6 +76,15 @@ public <T extends PlacementModifierType<?>> DeferredObject<T> registerPlacementM
         return deferredObject;
     }
 
+    @Override
+    public <T extends PoiType> DeferredObject<T> registerPoiType(ResourceLocation identifier, Supplier<T> supplier) {
+        return new DeferredObject<>(identifier, () -> {
+            T poiType = supplier.get();
+            Registry.register(BuiltInRegistries.POINT_OF_INTEREST_TYPE, identifier, poiType);
+            return poiType;
+        }).resolveImmediately();
```

Comment:
`PoiType` is managed in ForgeRegistries so it has to be registered via `DeferredRegisters` similar to `registerFeature`.